kolibri/plugins/setup_wizard/assets/src/views/setup-forms/default-language-form/index.vue

```diff
@@ -0,0 +1,124 @@
+<template>
+
+  <form @submit="setLanguage">
+    <fieldset class="default-language-form">
+      <legend>
+        <h1 class="default-language-form-header">
+          {{ $tr('languageFormHeader') }}
+        </h1>
+      </legend>
+
+      <label class="default-language-form-selected">
+        <span class="default-language-form-selected-label"> {{ $tr('selectedLanguageLabel') }} </span>
+        <span> {{ currentLanguage }} </span>
+      </label>
+
+      <k-button
+        v-for="language in buttonLanguages"
+        class="default-language-form-language-button"
+        :raised="false"
+        :text="language.name"/>
+
+      <label>
+        <span class="visuallyhidden">More Languages</span>
+        <select class="default-language-form-language-dropdown">
+          <option disabled selected> {{ $tr('showMoreLanguagesSelector') }} </option>
+          <option v-for="language in selectorLanguages" value="language.code">
+            {{ language.name }}
+          </option>
+        </select>
+      </label>
+
+      <k-button :primary="true" type="submit" :text="submitText" />
+    </fieldset>
+  </form>
+
+</template>
+
+
+<script>
+
+  import { availableLanguages as allLanguages, currentLanguage } from 'kolibri.utils.i18n';
+  import { submitDefaultLanguage } from '../../../state/actions/forms';
+  import kButton from 'kolibri.coreVue.components.kButton';
+  import omit from 'lodash/omit';
+
+  const numberOfLanguageButtons = 4;
+
+  // TODO add language switching logic
+
+  export default {
+    name: 'defaultLanguageForm',
+    $trs: {
+      languageFormHeader: 'Please select the default language for Kolibri',
+      showMoreLanguagesSelector: 'MORE',
+      selectedLanguageLabel: 'Selected',
+    },
+    components: { kButton },
+    props: {
+      submitText: {
+        type: String,
+        required: true,
+      },
+    },
+    data() {
```

Comment:
I am not actually sure if this makes a difference, but the Vue recommendation is that data should be a function, not an object method: https://vuejs.org/v2/api/#data

Reply:
Looks like we're actually doing it mostly wrong 😅 

Vue says "you should not use an arrow function". We have:

* 41 instances of `data: () =>`
* 0 instances of `data: function()`
* 27 instances of `data()`

Personally I tend to use the `data()` syntax - I believe but am not 100% sure that it's equivalent to the function syntax. However I'm fine with standardizing on either of the second two options.

Reply:
Well, the only reason you shouldn't use the arrow function is if you are referring to `this`, as long as you know not to do that, I think it's actually fine.

I guess given that the object that the method is for is the options object, and hence discarded after instantiation, the use of the method syntax is probably not harmful.

Reply:
One more item for the linter/consistency pile :)

Reply:
I think the linter dislikes `data: function()`, and demands `data()'